### PR TITLE
[JENKINS-38058] pass context when creating 'FolderItem' objects

### PR DESCRIPTION
**Pull request by mafraba**
[JENKINS-38058](https://issues.jenkins-ci.org/browse/JENKINS-38058)

Passing context page object to folders so that we can retrieve the real Jenkins instance later on

@reviewbybees

### CODE REVIEW

**Comment by raul-arabaolaza** [comment]
:bee:

**Comment by mafraba** [comment]
@reviewbybees done

File: src/main/java/org/jenkinsci/test/acceptance/po/JobsMixIn.java [hunk 1]
```diff
@@ -58,7 +58,14 @@ public Object call() throws Exception {
     }
 
     public <T extends TopLevelItem> T get(Class<T> type, String name) {
-        return newInstance(type, injector, url("job/%s/", name), name);
+        try {
+            assert getContext() != null;
+            return newInstance(type, getContext(), url("job/%s/", name), name);
+        } catch (AssertionError e) {
```
Comment (olivergondza) [inline src/main/java/org/jenkinsci/test/acceptance/po/JobsMixIn.java]: I do not like this as we throw the exception all over the place. We should either fix all the implementations and accept external client will be broken or make the `newInstance` to throw more specific exception to implement this fallback.

Comment (mafraba) [inline src/main/java/org/jenkinsci/test/acceptance/po/JobsMixIn.java]: Fair enough. I propose a different solution with an explicit check for the constructor, please see new commit.

**Pull request merged**